src/Compiler.Dynamic/DynamicControlCollection.cs

```diff
@@ -87,12 +88,44 @@ void IDisposable.Dispose()
 
     Type? ITypeResolutionService.GetType(string name, bool throwOnError)
     {
-        throw new NotImplementedException();
+        return GetType(name, throwOnError, ignoreCase: false);
+    }
+
+    private static Type? GetType(string typeName, bool throwOnError, bool ignoreCase)
+    {
+        Type? type = null;
+        if (Util.TypeNameContainsAssembly(typeName))
+        {
+            type = Type.GetType(typeName, throwOnError, ignoreCase);
+
+            if (type != null)
+            {
+                return type;
+            }
+        }
+
+        if(type == null)
```

Comment:
Taylor, this would be expensive (while in old System.Web also it would take same time or less). But if there is a better way, do let me know.. we can take a follow-up task or incorporate in this one. Probably this is the most imp piece in this PR (where I am looking some guidance)

Reply:
It's fine for now. If we need to, we can cache it somehow so it's not necessary if we're doing the same lookup a lot